src/modules/post/graphql/mutation/post_create.rs

```diff
@@ -28,9 +28,16 @@ impl PostCreate {
         let token = auth.token()?;
         let user = services.auth.whoami(token).await?;
 
-        match services.post.create(user, input).await {
+        match services.post.create(user.clone(), input).await {
             Ok(post) => Ok(PostCreate {
-                post: Some(post),
+                post: Some(Post {
+                    id: post.id,
+                    content: post.content,
+                    user,
+                    scope: post.scope,
+                    created_at: post.created_at,
+                    updated_at: post.updated_at,
+                }),
```

Comment:
Another good refactor would be having a factory function for this. Domain Entity <-> GraphQL Type conversion